src/lib/agent.c

```diff
@@ -67,25 +65,24 @@ void sa_init(struct sof *sof, uint64_t timeout)
 
 	trace_sa("sa_init(), timeout = %u", timeout);
 
-	sa = rzalloc(SOF_MEM_ZONE_SYS, 0, SOF_MEM_CAPS_RAM, sizeof(*sa));
-	sof->sa = sa;
+	sof->sa = rzalloc(SOF_MEM_ZONE_SYS, 0, SOF_MEM_CAPS_RAM,
```

Comment:
I assume this structure either was already there or was added by some other patch?

Reply:
Yes, it was here before.